win-odbc64/odbcclient/drvr35/sqltocconv.cpp

```diff
@@ -36,6 +36,34 @@
 
 #define MAXCHARLEN 32768 //32K
 
+// for server2008 when using function pow() then throws STATUS_ILLEGAL_INSTRUCTION
+__int64 pow(int base, short scale)
+{
+	DWORD dwVersion = 0;
+	DWORD dwBuild = 0;
+
+	dwVersion = GetVersion();
+
+	// Get the build number.
+
+	if (dwVersion < 0x80000000)
+		dwBuild = (DWORD)(HIWORD(dwVersion));
+
+	__int64 retValue = 1;
+	if (dwBuild == 7600)
+	{
+		scale = scale > 18 ? 18 : scale;
+		for (int i = 0; i < scale; i++)
+			retValue = retValue * 10;
+	}
+	else
+	{
+		retValue = pow((double)base, scale);

```

Comment:
Suggest to look at the errno to return proper errors.  I wonder due to compatible method signature, there is possibility that linker might resolve to the standard pow function instead of calling this function.

Reply:
When using the standard pow function in ODBC under server2008, it throws STATUS_ILLEGAL_INSTRUCTION.
For the standard pow function, the type of parameter base is _Common_float_type including float, double and long double.
So using type int  as the parameter in pow() can lead linker to call this function first.

Reply:
I would suggest to retain double data type for return type to enable a large result to be accommodated without raising numeric overflow. Hence, I would suggest Trafodion function can have an extra parameter like to pass the errno as in
double pow (double base, double exponent, int *error)

Pass the errno to be returned to the caller for it handle it. This would also take care of any ambiguity in resolution of the method

Reply:
I have token the advice, can you review it if you are free.